### PR TITLE
Add Slayer Task Locations (Konar)

### LINKED ISSUE #7267: Tasks from konar don't trigger the slayer counter to show.

### PR DESCRIPTION
**Pull request by ChaoticConundrum**
Fix some pattern matching for new slayer game text, and record task locations. Locations are also stored in config, and displayed on the task counter:

capture2
(ignore the zero points/streak, the config was somehow wiped out, unrelated to this PR)

This does **NOT** add the new slayer monsters to the plugin, I will leave it to someone else to pick the icons for that.

The game text currently understood:
- Gem 'Check" progress (same as slayer helmet?)
- Konar dialogue for new task
- Konar dialogue with current Konar task
- Konar dialogue with current non-Konar task
- Slayer master dialogue with current Konar task

I don't know what the game text is for Konar tasks with:
- Receiving a task from a slayer partner

Testing this is slow, so help testing would be appreciated.

This would resolve #7267.

### CODE REVIEW

**Comment by ChaoticConundrum** [comment]
Apparently tests are broken, working on it.

**Comment by Adam-** [comment]
This should be extending the test with the newer messages too.

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 1]
```diff
@@ -604,10 +614,20 @@ private void rebuildTargetList()
...
+	}
+
+	private void setTask(String name, int amt, int initAmt, String location)
 	{
 		taskName = name;
 		amount = amt;
 		initialAmount = initAmt;
+		if (location == null)
+		{
+			location = "";
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Why?

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java [hunk 2]
```diff
@@ -140,12 +140,20 @@ default int amount()
 		return -1;
 	}
 
+	@ConfigItem(
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java]: Why was this moved?

Comment (ChaoticConundrum) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerConfig.java]: Just to reorganize each config item into getter/setter pairs, at some point another item was added between the getter/setter for that item. Of course it's not essential, but I was editing right next to it anyway.

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java (hunk 1 above)
Comment (ChaoticConundrum) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: To simplify the logic where Matcher.group(String) is used, for the Pattern group that selects the task location, which is not always present. Matcher.group(String) returns null if the group is not present, so I sanitize null into the empty string here.

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 3]
```diff
@@ -98,10 +98,10 @@
 	private static final Pattern COMBAT_BRACELET_TASK_UPDATE_MESSAGE = Pattern.compile("^You still need to kill (\\d+) monsters to complete your current Slayer assignment");
 
 	//NPC messages
-	private static final Pattern NPC_ASSIGN_MESSAGE = Pattern.compile(".*Your new task is to kill\\s*(\\d*) (.*)\\.");
+	private static final Pattern NPC_ASSIGN_MESSAGE = Pattern.compile(".*(?:Your new task is to kill|You are to bring balance to)\\s*(?<amount>\\d+) (?<name>.+?)(?: in the (?<location>.+))?\\.");
```
[the remark this answers is not in the document]
Comment (ChaoticConundrum) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: You are right, good catch.

**Comment by ChaoticConundrum** [comment]
From the examples I have seen, the Krystilia tasks should be correctly captured. I don't do wilderness slayer, so my experience is limited.

**Comment by ChaoticConundrum** [comment]
I still don't have the text for receiving a Konar task from a slayer partner, in case anyone comes across it.

**Comment by ChaoticConundrum** [comment]
I thought PR commits were squashed?

**Comment by deathbeam** [comment]
They are squashed if they are doing only 1 thing
